src/commons/resources.rs

```diff
@@ -318,6 +319,107 @@ impl<T, K> Into<ResourceRequirements> for Resources<T, K> {
+pub struct ResourceRequirementsPolicyError {
+    resource_policy: ResourceRequirementsPolicy,
+    container_name: String,
+    resource_key: String,
+}
+
+/// [`ResourceRequirementsPolicy`] dsscribes the available resource requirement
+/// policies. The user can set limits, requests and claims. This enum makes it
+/// possible to check if containers set one or more of these policies.
+#[derive(Copy, Clone, Debug, Display)]
+#[strum(serialize_all = "lowercase")]
+pub enum ResourceRequirementsPolicy {
+    Limits,
+    Requests,
+    Claims,
+}
+
+pub trait ResourceRequirementsPolicyExt {
+    fn check_policy_for_resource(
+        &self,
+        policy: ResourceRequirementsPolicy,
+        resource: &str,
+    ) -> Result<(), ResourceRequirementsPolicyError>;
+
+    fn check_policies_for_resource(
+        &self,
+        policies: Vec<ResourceRequirementsPolicy>,
+        resource: &str,
+    ) -> Result<(), ResourceRequirementsPolicyError>;
+}
+
+impl ResourceRequirementsPolicyExt for Container {
+    fn check_policy_for_resource(
+        &self,
+        policy: ResourceRequirementsPolicy,
+        resource: &str,
+    ) -> Result<(), ResourceRequirementsPolicyError> {
+        let rr = self
+            .resources
+            .as_ref()
+            .ok_or(ResourceRequirementsPolicyError {
+                container_name: self.name.clone(),
+                resource_key: resource.into(),
+                resource_policy: policy,
+            })?;
+
+        match policy {
+            ResourceRequirementsPolicy::Limits => {
+                let limits = rr.limits.as_ref().ok_or(ResourceRequirementsPolicyError {
+                    container_name: self.name.clone(),
+                    resource_key: resource.into(),
+                    resource_policy: policy,
+                })?;
+
+                if !limits.contains_key(resource) {
+                    return Err(ResourceRequirementsPolicyError {
+                        container_name: self.name.clone(),
+                        resource_key: resource.into(),
+                        resource_policy: policy,
+                    });
+                }
+            }
+            ResourceRequirementsPolicy::Requests => {
+                let requests = rr
+                    .requests
+                    .as_ref()
+                    .ok_or(ResourceRequirementsPolicyError {
+                        container_name: self.name.clone(),
+                        resource_key: resource.into(),
+                        resource_policy: policy,
+                    })?;
+
+                if !requests.contains_key(resource) {
+                    return Err(ResourceRequirementsPolicyError {
+                        container_name: self.name.clone(),
+                        resource_key: resource.into(),
+                        resource_policy: policy,
+                    });
+                }
+            }
+            ResourceRequirementsPolicy::Claims => todo!(),
```

Comment:
Marker to not forget.

Reply:
How do we handle claims? Do we need to do any validation here?

Reply:
Nope, I'm not aware of any claim usages. I would suggest to not support it for now

Reply:
So should we remove it from the enum completely or should we keep the `todo!()` macro (or print out an error)

Reply:
i would remove the enum variant

Reply:
Removed in d48d06c